src/helpers/credential.rs

```diff
@@ -269,6 +276,152 @@ impl<'a> Credential<'a> {
         Ok(())
     }
 
+    pub async fn create_self_issued_credential(
+        &mut self,
+        schema_did: &str,
+        credential_subject_str: &str,
+        bbs_secret: &str,
+        bbs_private_key: &str,
+        credential_revocation_did: Option<&str>,
+        credential_revocation_id: Option<&str>,
+        exp_date: Option<&str>,
+    ) -> Result<String, CredentialError> {
+        let credential_subject: CredentialSubject = serde_json::from_str(credential_subject_str)?;
+        let subject_did = credential_subject.clone().id.ok_or("no subject did found".to_string()).unwrap();
+        let subject_did_str = subject_did.as_str();
+        let schema: CredentialSchema = self.get_did_document(schema_did).await?;
+        let issuer_public_key = self
+            .get_issuer_public_key(&subject_did, "#bbs-key-1")
+            .await?;
+        let use_valid_until= if exp_date.is_some() { true } else { false };
+        let exp_date_str = exp_date.unwrap_or("");
```

Comment:
Can we move the `unwrap_or` behaviour into the `let valid_until = if use_valid_until ` block below? If placed here, it looks like we'd actually be using the empty string as the value for `valid_until` but `exp_date_str` is unused if `use_valid_until` is `false`. ^^

Reply:
Done, thank you!